src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubScaleMonitor.cs

```diff
@@ -0,0 +1,395 @@
+        private const int PartitionLogIntervalInMinutes = 5;
+
+        private readonly string _functionId;
+        private readonly string _eventHubName;
+        private readonly string _consumerGroup;
+        private readonly EventHubsConnectionStringBuilder _connectionStringBuilder;
+        private readonly CloudBlobContainer _blobContainer;
+        private readonly ScaleMonitorDescriptor _scaleMonitorDescriptor;
+        private readonly ILogger _logger;
+
+        private EventHubClient _client;
+        private DateTime _nextPartitionLogTime;
+        private DateTime _nextPartitionWarningTime;
+
+        public EventHubScaleMonitor(
+            string functionId,
+            string eventHubName,
+            string consumerGroup,
+            string connectionString,
+            string storageConnectionString,
+            ILogger logger,
+            CloudBlobContainer blobContainer = null)
+        {
+            _functionId = functionId;
+            _eventHubName = eventHubName;
+            _consumerGroup = consumerGroup;
+            _logger = logger;
+            _scaleMonitorDescriptor = new ScaleMonitorDescriptor($"{_functionId}-EventHubTrigger-{_eventHubName}".ToLower());
+            _nextPartitionLogTime = DateTime.UtcNow;
+            _nextPartitionWarningTime = DateTime.UtcNow;
+
+            _connectionStringBuilder = new EventHubsConnectionStringBuilder(connectionString);
+            if (!String.IsNullOrEmpty(_eventHubName))
+            {
+                _connectionStringBuilder.EntityPath = _eventHubName;
+            }
+            _client = EventHubClient.CreateFromConnectionString(_connectionStringBuilder.ToString());
+
+            _blobContainer = blobContainer;
+            if (_blobContainer == null)
+            {
+                CloudStorageAccount cloudStorageAccount = CloudStorageAccount.Parse(storageConnectionString);
+                CloudBlobClient blobClient = cloudStorageAccount.CreateCloudBlobClient();
+                _blobContainer = blobClient.GetContainerReference(EventHubContainerName);
+            }
+        }
+
+        public ScaleMonitorDescriptor Descriptor
+        {
+            get
+            {
+                return _scaleMonitorDescriptor;
+            }
+        }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await GetMetricsAsync();
+        }
+
+        public async Task<EventHubTriggerMetrics> GetMetricsAsync()
+        {
+            EventHubTriggerMetrics metrics = new EventHubTriggerMetrics();
+            EventHubRuntimeInformation runtimeInfo = null;
+
+            try
+            {
+                runtimeInfo = await _client.GetRuntimeInformationAsync();
+            }
+            catch (NotSupportedException e)
+            {
+                _logger.LogWarning($"Event Hub Trigger does not support Notification Hubs. Error: {e.Message}");

```

Comment:
Is this error handling logic that you just copied from ScaleController? Seems odd to me that we know this is Notification Hubs specific here.

Also, nit: EventHub/NotificationHubs

Reply:
Yeah I'm not sure how that determination is made, but yes it's straight from ScaleController

Reply:
I think it's bogus and should be fixed, but we can do that another time. Good enough to just carry the logic forward for now.